src/backend/distributed/worker/worker_partition_protocol.c

```diff
@@ -2,1304 +2,42 @@
+ * worker_range_partition_table is a deprecated function that we keep around for
+ * testing downgrade scripts.
  */
 Datum
 worker_range_partition_table(PG_FUNCTION_ARGS)
 {
-	CheckCitusVersion(ERROR);
-
-	uint64 jobId = PG_GETARG_INT64(0);
-	uint32 taskId = PG_GETARG_UINT32(1);
-	text *filterQueryText = PG_GETARG_TEXT_P(2);
-	text *partitionColumnText = PG_GETARG_TEXT_P(3);
-	char *partitionColumnNameCandidate = text_to_cstring(partitionColumnText);
-
-	char *partitionColumnName = NULL;
-	uint32 partitionColumnIndex = 0;
-	PartitionColumnIndexOrPartitionColumnName(partitionColumnNameCandidate,
-											  &partitionColumnName,
-											  &partitionColumnIndex);
-
-	Oid partitionColumnType = PG_GETARG_OID(4);
-	ArrayType *splitPointObject = PG_GETARG_ARRAYTYPE_P(5);
-
-
-	const char *filterQuery = text_to_cstring(filterQueryText);
-
-	/* first check that array element's and partition column's types match */
-	Oid splitPointType = ARR_ELEMTYPE(splitPointObject);
-
-	if (splitPointType != partitionColumnType)
-	{
-		ereport(ERROR, (errmsg("partition column type %u and split point type %u "
-							   "do not match", partitionColumnType, splitPointType)));
-	}
-
-	/* use column's type information to get the comparison function */
-	FmgrInfo *comparisonFunction = GetFunctionInfo(partitionColumnType,
-												   BTREE_AM_OID, BTORDER_PROC);
-
-	/* deserialize split points into their array representation */
-	Datum *splitPointArray = DeconstructArrayObject(splitPointObject);
-	int32 splitPointCount = ArrayObjectCount(splitPointObject);
-	uint32 fileCount = splitPointCount + 1; /* range partitioning needs an extra bucket */
-
-	/* create range partition context object */
-	RangePartitionContext *partitionContext = palloc0(sizeof(RangePartitionContext));
-	partitionContext->comparisonFunction = comparisonFunction;
-	partitionContext->splitPointArray = splitPointArray;
-	partitionContext->splitPointCount = splitPointCount;
-
-	/* init directories and files to write the partitioned data to */
-	StringInfo taskDirectory = InitTaskDirectory(jobId, taskId);
-	StringInfo taskAttemptDirectory = InitTaskAttemptDirectory(jobId, taskId);
-
-	FileOutputStream *partitionFileArray = OpenPartitionFiles(taskAttemptDirectory,
-															  fileCount);
-	FileBufferSizeInBytes = FileBufferSize(PartitionBufferSize, fileCount);
-
-	/* call the partitioning function that does the actual work */
-	FilterAndPartitionTable(filterQuery, partitionColumnName, partitionColumnIndex,
-							partitionColumnType,
-							&RangePartitionId, (const void *) partitionContext,
-							partitionFileArray, fileCount);
-
-	/* close partition files and atomically rename (commit) them */
-	ClosePartitionFiles(partitionFileArray, fileCount);
-	CitusRemoveDirectory(taskDirectory->data);
-	RenameDirectory(taskAttemptDirectory, taskDirectory);
-	PG_RETURN_VOID();
+	ereport(ERROR, (errmsg("this function is deprecated and only kept for testing "
```

Comment:
Minor: if that's simple enough, maybe we move the 8 UDFs you remove to single file like `worker_partition_protocol` or such? 

Given that we won't be dropping these soon, could be nice to document all there. Obviously optional suggestion